File: contest/j.py
import math as m

while True:
    try:
        n = int(input())
        max = int(m.sqrt(n))
        flag = False
        for i in range(max + 1, 0, -1):
            for j in range(max + 1):
                #print("==", (i ** 2) + (j ** 2), "==")
                if flag:
                    break
                elif ((i ** 2) + (j ** 2)) == n:
                    #print("==(", i,j,")==", (i ** 2) + (j ** 2), "==")
                    print(1)
                    flag = True
                    break
        if not flag:
            print(0)
    except EOFError:
        break

File: contest/j.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    long long unsigned n;
    while(cin >> n){
        long unsigned max = ((long unsigned) sqrt(n)) + 1;
        bool flag = false;
        for(long unsigned i = max; i > 0;i--){
            for(long unsigned j = 1; j < max;j++){
                if(flag) break;
                else if(pow(i, 2) + pow(j, 2) == n){
                    cout << "1" << endl;
                    flag = true;
                    break;
                }
            }   
        }
        if(!flag) cout << "0" << endl;
    }
}